x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/utils.ts

```diff
@@ -1027,3 +1050,83 @@ export const getDisabledActionsWarningText = ({
+  (alertObject as EqlShellAlert800)?.[ALERT_UUID] != null &&
+  (alertObject as EqlShellAlert800)?.[ALERT_GROUP_ID] != null;
+
+type RuleWithInMemorySuppression = ThreatRuleParams | EqlRuleParams | MachineLearningRuleParams;
+
+export interface SequenceSuppressionTermsAndFieldsParams {
+  shellAlert: WrappedEqlShellOptionalSubAlertsType;
+  buildingBlockAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+  spaceId: string;
+  completeRule: CompleteRule<RuleWithInMemorySuppression>;
+  mergeStrategy: ConfigType['alertMergeStrategy'];
+  indicesToQuery: string[];
+  buildReasonMessage: BuildReasonMessage;
+  alertTimestampOverride: Date | undefined;
+  ruleExecutionLogger: IRuleExecutionLogForExecutors;
+  publicBaseUrl: string | undefined;
+  primaryTimestamp: string;
+  secondaryTimestamp?: string;
+}
+
+export type SequenceSuppressionTermsAndFieldsFactory = (
+  shellAlert: WrappedEqlShellOptionalSubAlertsType,
+  buildingBlockAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>,
+  buildReasonMessage: BuildReasonMessage
+) => WrappedFieldsLatest<EqlShellFieldsLatest & SuppressionFieldsLatest> & {
+  subAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+};
+
+export const sequenceSuppressionTermsAndFieldsFactory = ({
+  shellAlert,
+  buildingBlockAlerts,
+  spaceId,
+  completeRule,
+  mergeStrategy,
+  indicesToQuery,
+  buildReasonMessage,
+  alertTimestampOverride,
+  ruleExecutionLogger,
+  publicBaseUrl,
+  primaryTimestamp,
+  secondaryTimestamp,
+}: SequenceSuppressionTermsAndFieldsParams): WrappedFieldsLatest<
+  EqlShellFieldsLatest & SuppressionFieldsLatest
+> & {
+  subAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+} => {
+  console.error('ARE WE IN HERE');
+  const suppressionTerms = getSuppressionTerms({
+    alertSuppression: completeRule?.ruleParams?.alertSuppression,
+    fields: shellAlert?._source,
+  });
+  console.error('SUPPRESSION TERMS', suppressionTerms);
+  const instanceId = objectHash([suppressionTerms, completeRule.alertId, spaceId]);
+
+  const suppressionFields = getSuppressionAlertFields({
+    primaryTimestamp,
+    secondaryTimestamp,
+    // as casting should work because the alert fields are flattened (hopefully?)
+    fields: shellAlert?._source as Record<string, string | number | null> | undefined,
+    suppressionTerms,
+    fallbackTimestamp: alertTimestampOverride?.toISOString() ?? new Date().toISOString(),
+    instanceId,
+  });
+  console.error('SUPPRESSION FIELDS', JSON.stringify(suppressionFields, null, 2));
+  const theFields = Object.keys(suppressionFields) as Array<keyof ExtraFieldsForShellAlert>;
+  // mutates shell alert to contain values from suppression fields
+  theFields.forEach((field) => (shellAlert._source[field] = suppressionFields[field]));
+  shellAlert.subAlerts = buildingBlockAlerts;
+
+  return shellAlert as WrappedFieldsLatest<EqlShellFieldsLatest & SuppressionFieldsLatest> & {
```

Comment:
We should create the `shellAlert` in such a way that casting isn't necessary

Reply:
Fixed here: https://github.com/elastic/kibana/pull/189725/commits/3079b9eabcd2274281f1678704f1b09c5cc0e655